internal/common/conversion/schema_description.go

```diff
@@ -2,15 +2,83 @@ package conversion
 
 import (
 	"reflect"
+	"slices"
 
 	dsschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 )
 
+func DataSourceSchemaFromResource(rs schema.Schema, requiredFields ...string) dsschema.Schema {
```

Comment:
q: Are there any limitations worth calling out? I do see that for example project data source does have optional attributes which would not be supported.

Reply:
at the moment blocks is not implemented (it panics so it's evident that needs to be implemented).  my idea is to add functionality as we need it.  Every time we use this func for existing data source schemas, we need to be sure that we're generating the same schema, as I did before this commit https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2799/commits/5c6bd8b743af5ad30c82bc288386e935327bf051, and implement any additional logic if needed.